### PR TITLE
Add an example for macro classes

### PR DESCRIPTION
**Pull request by j8r**
Macros can be also used to generate classes and structs.

### CODE REVIEW

File: syntax_and_semantics/macros.md [hunk 1]
```diff
@@ -94,6 +94,39 @@ end
...
+end
+
+# This generates:
+#     module Foo
+#       class Bar
+#         def initialize(@name : String)
+#         end
+#
+#         def say
+#           "hi " + @name
```
Comment (GloverDonovan) [inline syntax_and_semantics/macros.md]: `"hi #{@name}"` would probably be better here.

**Comment by j8r** [comment]
I usually use a Markdown previewer, but I'm quite on the go this time and miss it.

**Comment by j8r** [comment]
Can it be merged?

**Comment by j8r** [comment]
Ping – nothing new?

**Pull request merged**